x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -713,7 +714,9 @@ public void onStackFramesResponse(MultiGetResponse multiGetItemResponses) {
                     if (stackFrames.containsKey(frame.getId()) == false) {
                         StackFrame stackFrame = StackFrame.fromSource(frame.getResponse().getSource());
                         if (stackFrame.isEmpty() == false) {
-                            stackFrames.putIfAbsent(frame.getId(), stackFrame);
+                            if (stackFrames.putIfAbsent(frame.getId(), stackFrame) == null) {
+                                totalInlineFrames.addAndGet(stackFrame.size() - 1); // -1 to only count inlined frames
```

Comment:
Can you please move this code into a method called `inlineFrameCount` (or the like) in `StackFrame` and use that one here instead? That would clarify the intent of this code IMHO.